[BUG] Documents no longer open

### Description

When I open a document, assign a tag and click Save and Next, it doesn't always happen, but very often, that the next document doesn't open. It just says “loading” on the right. All links on Paperless are not clickable and the page cannot be reloaded. The only solution is to close and reopen the Safari browser or open a new tab.

### Steps to reproduce

I have a filter with TAG "INBOX" where I currently have around 1,000 documents. The documents are sorted alphabetically in ascending order.
1.) I open the 1st document
2.) I assign an existing TAG
3.) Click on Save and Continue

Next document opens, or not.
Sometimes it works for a few documents, sometimes it happens immediately with the next document.

When evaluating the logs, I saw that the emails were being checked at the same time. Is there perhaps a connection?

### Webserver logs

```bash
[2024-01-25 08:50:01,744] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[9aa475a7-3857-4f5b-925c-87f1033b12cf] succeeded in 1.6577134882099926s: 'No new documents were added.'
[2024-01-25 09:00:00,096] [INFO] [celery.beat] Scheduler: Sending due task Check all e-mail accounts (paperless_mail.tasks.process_mail_accounts)
[2024-01-25 09:00:00,100] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[78650b55-82f9-441e-8918-0781d5b669e9] received
[2024-01-25 09:00:01,280] [ERROR] [paperless_mail] Rule Filearchiv.Fileablage: Error while processing rule: date value out of range
Traceback (most recent call last):
  File "/usr/src/paperless/src/paperless_mail/mail.py", line 514, in handle_mail_account
    total_processed_files += self._handle_mail_rule(
                             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/paperless_mail/mail.py", line 561, in _handle_mail_rule
    criterias = make_criterias(rule, supports_gmail_labels)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/paperless_mail/mail.py", line 372, in make_criterias
    maximum_age = date.today() - timedelta(days=rule.maximum_age)
                  ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
OverflowError: date value out of range
[2024-01-25 09:00:01,903] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[78650b55-82f9-441e-8918-0781d5b669e9] succeeded in 1.7995521272532642s: 'No new documents were added.'
```


### Browser logs

_No response_

### Paperless-ngx version

2.4.1

### Host OS

linux x86_64 Synology NAS

### Installation method

Docker - official image

### Browser

Safari

### Configuration changes

No

### Other

_No response_

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

--- a/src-ui/src/app/components/document-detail/document-detail.component.html
+++ b/src-ui/src/app/components/document-detail/document-detail.component.html
@@ -324,44 +324,45 @@
         <ng-container i18n>Loading...</ng-container>
       </div>
     </div>
-  }
-  @switch (contentRenderType) {
-    @case (ContentRenderType.PDF) {
-      @if (!useNativePdfViewer) {
-        <div class="preview-sticky pdf-viewer-container">
-          <pngx-pdf-viewer
-            [src]="{ url: previewUrl, password: password }"
-            [original-size]="false"
-            [show-borders]="true"
-            [show-all]="true"
-            [(page)]="previewCurrentPage"
-            [zoom-scale]="previewZoomScale"
-            [zoom]="previewZoomSetting"
-            (error)="onError($event)"
-            (after-load-complete)="pdfPreviewLoaded($event)">
-          </pngx-pdf-viewer>
+  } @else {
+    @switch (contentRenderType) {
+      @case (ContentRenderType.PDF) {
+        @if (!useNativePdfViewer) {
+          <div class="preview-sticky pdf-viewer-container">
+            <pngx-pdf-viewer
+              [src]="{ url: previewUrl, password: password }"
+              [original-size]="false"
+              [show-borders]="true"
+              [show-all]="true"
+              [(page)]="previewCurrentPage"
+              [zoom-scale]="previewZoomScale"
+              [zoom]="previewZoomSetting"
+              (error)="onError($event)"
+              (after-load-complete)="pdfPreviewLoaded($event)">
+            </pngx-pdf-viewer>
+          </div>
+        } @else {
+          <object [data]="previewUrl | safeUrl" class="preview-sticky" width="100%"></object>
+        }
+      }
+      @case (ContentRenderType.Text) {
+        <div class="preview-sticky bg-light p-3 overflow-auto" width="100%">{{previewText}}</div>
+      }
+      @case (ContentRenderType.Image) {
+        <div class="preview-sticky">
+          <img [src]="previewUrl | safeUrl" width="100%" height="100%" alt="{{title}}" />
         </div>
-      } @else {
+      }
+      @case (ContentRenderType.Other) {
         <object [data]="previewUrl | safeUrl" class="preview-sticky" width="100%"></object>
       }
     }
-    @case (ContentRenderType.Text) {
-      <div class="preview-sticky bg-light p-3 overflow-auto" width="100%">{{previewText}}</div>
-    }
-    @case (ContentRenderType.Image) {
-      <div class="preview-sticky">
-        <img [src]="previewUrl | safeUrl" width="100%" height="100%" alt="{{title}}" />
+    @if (requiresPassword) {
+      <div class="password-prompt">
+        <form>
+          <input autocomplete="" autofocus="true" class="form-control" i18n-placeholder placeholder="Enter Password" type="password" (keyup)="onPasswordKeyUp($event)" />
+        </form>
       </div>
     }
-    @case (ContentRenderType.Other) {
-      <object [data]="previewUrl | safeUrl" class="preview-sticky" width="100%"></object>
-    }
-  }
-  @if (requiresPassword) {
-    <div class="password-prompt">
-      <form>
-        <input autocomplete="" autofocus="true" class="form-control" i18n-placeholder placeholder="Enter Password" type="password" (keyup)="onPasswordKeyUp($event)" />
-      </form>
-    </div>
   }
 </ng-template>

--- a/src-ui/src/app/components/document-detail/document-detail.component.spec.ts
+++ b/src-ui/src/app/components/document-detail/document-detail.component.spec.ts
@@ -681,6 +681,7 @@ describe('DocumentDetailComponent', () => {
 
   it('should support Enter key in password field', () => {
     initNormally()
+    component.metadata = { has_archive_version: true }
     component.onError({ name: 'PasswordException' }) // normally dispatched by pdf viewer
     fixture.detectChanges()
     expect(component.password).toBeUndefined()

--- a/src-ui/src/app/components/document-detail/document-detail.component.ts
+++ b/src-ui/src/app/components/document-detail/document-detail.component.ts
@@ -82,6 +82,7 @@ enum ContentRenderType {
   Image = 'image',
   Text = 'text',
   Other = 'other',
+  Unknown = 'unknown',
 }
 
 enum ZoomSetting {
@@ -211,6 +212,7 @@ export class DocumentDetailComponent
   }
 
   get contentRenderType(): ContentRenderType {
+    if (!this.metadata) return ContentRenderType.Unknown
     const contentType = this.metadata?.has_archive_version
       ? 'application/pdf'
       : this.metadata?.original_mime_type

